core/build.gradle.kts

```diff
@@ -0,0 +1,114 @@
+import com.saveourtool.processbuilder.configureDetekt
+import com.saveourtool.processbuilder.configureDiktat
+
+plugins {
+    kotlin("multiplatform")
+}
+
+kotlin {
+    jvm {
+        jvmToolchain(11)
+        withJava()
+        testRuns["test"].executionTask.configure {
+            useJUnitPlatform()
+        }
+    }
+    js(IR) {
+        browser {
+            commonWebpackConfig {
+                cssSupport {
+                    enabled.set(true)
+                }
+            }
+        }
+    }
+
+    linuxX64()
+    macosX64()
+    mingwX64()
+
+    sourceSets {
+        all {
+            languageSettings.optIn("kotlin.RequiresOptIn")
+        }
+
+        val commonMain by getting {
+            dependencies {
+                api(libs.okio)
+
+                implementation(libs.kotlin.logging)
+                implementation(libs.kotlinx.datetime)
+                implementation(libs.kotlinx.coroutines.core)
+            }
+        }
+
+        val commonTest by getting
+
+        /**
+         * js is supported in order to allow using ProcessBuilder in common section
+         * js tests make no sense so common tests are targeted as commonNonJsTest
+         * commonNonJsMain is required in order to make commonNonJsTest resolvable
+         */
+        val commonNonJsMain by creating {
+            dependsOn(commonMain)
+        }
+
+        val commonNonJsTest by creating {
+            dependsOn(commonTest)
+            dependencies {
+                implementation(libs.okio.fakefilesystem)
+                implementation(libs.kotest.assertions.core)
+                implementation(kotlin("test-common"))
+                implementation(kotlin("test-annotations-common"))
+            }
+        }
+
+        val jvmMain by getting {
+            dependsOn(commonNonJsMain)
+            dependencies {
+                implementation(libs.slf4j)
+            }
+        }
+
+        val jvmTest by getting {
+            dependsOn(commonNonJsTest)
+            dependencies {
+                implementation(kotlin("test-junit5"))
+                implementation(libs.junit.jupiter.engine)
+            }
+        }
+
+        val jsMain by getting {
```

Comment:
btw, if we doesn't support JS, maybe just do no register such sourceSet?

Reply:
The idea was to make ProcessBuilder be able to be imported to `commonMain`

Moreover, js implementation is possible for `nodeJs`, so I would like to leave as is

Reply:
> js implementation is possible for `nodeJs`

We can also do not provide source set for `nodeJs`. I don't see a reason to provide implementation which just throw exception for all cases without any plans to implement it in someday.

are there any technical restrictions that force us to register `js`?

Reply:
Done